src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -844,7 +854,20 @@ public override ValueTask DisposeAsync()
 #endif
         {
             using (NoSynchronizationContextScope.Enter())
-                return new ValueTask(Close(connectionClosing: false, async: true));
+                return DisposeAsyncCore();
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            async ValueTask DisposeAsyncCore()
+            {
+                try
+                {
+                    await Close(connectionClosing: false, async: true);
+                }
+                catch (Exception e)
+                {
+                    Log.Error("Exception caught while disposing a reader", e, Connector.Id);
+                }
+            }
```

Comment:
Otherwise, you can add the code at a single place. I mean in the internal `Close`.

Reply:
Close should not swallow exceptions